internal/store_test.go

```diff
@@ -201,3 +203,92 @@ func TestStore_SinkWritePolicyWeight(t *testing.T) {
 	require.Equal(t, 8, int(store.policy.weightedSize))
 
 }
+
+func TestStore_CloseRace(t *testing.T) {
+	store := NewStore[int, int](1000, false, true, nil, nil, nil, 0, 0, nil)
```

Comment:
Not for this PR, but holy cow that is a lot of parameters 🤣 are all of those mandatory?! If they aren't, I suggest using the "functional options pattern" described [here](https://golang.cafe/blog/golang-functional-options-pattern.html). My team has been using that with quite a lot of success.